### PR TITLE
Updates to example PSE for C26

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
# Tickets
- [Example PSE Ticket](https://app.asana.com/1/181459410160484/project/1205655776549324/task/1215168137850940?focus=true)
- [Parent Ticket](https://app.asana.com/1/181459410160484/project/1205655776549324/task/1212367525684514?focus=true)

# Related Gist Review
Since we updated the logical step review template, I needed to update the example conversation we link to in the explanation section. Since gists don't support PRs, the new version is in a secret gist [here for review](https://gist.github.com/ada-instructors/997ce79959113865bb4df077ff50d400) and the old version is [here](https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1) for comparison.
- Once reviewed/any updates made to the temp gist, I'll copy the contents over to the original gist and delete the temp so we don't need to update the link in the lesson.

# Ask for Feedback
- We wanted to start including placeholders for the Questions & Observations section using examples from the example PSE on future PSEs. For this one, I didn't know if we also wanted some kind of placeholder, so I added something, would love feedback if folks think that's useful or have suggestions for improvements.
- I also added a sentence about test naming to the writing tests step and would love to hear if folks think that's helpful or should be removed.

### CODE REVIEW

File: about-pses/example-pse.md [hunk 1]
```diff
@@ -56,6 +56,17 @@ As you come up with observations and questions, assume that error handling for i
 - String Sanitization: You may assume strings will not include invalid characters.
 
 ##### !end-question
+##### !placeholder
+
+Add observations, assuptions, and questions here:
```
Comment (yangashley) [inline about-pses/example-pse.md]:
```suggestion
Add observations, assumptions, and questions here:
```

File: about-pses/example-pse.md [hunk 2]
```diff
@@ -56,6 +56,17 @@ As you come up with observations and questions, assume that error handling for i
...
+
+1. From the problem description, I observe that...
+2. From the example inputs I can see...
+3. A question I have around inputs is...
+4. ...
+5. ...
```
Comment (yangashley) [inline about-pses/example-pse.md]: These placeholders look good to me 👍 

Some other potential sentence starters could be:
- "Can I assume that...?"
- "What behavior is expected when...?"

File: about-pses/example-pse.md [hunk 3]
```diff
@@ -229,6 +154,9 @@ One helpful insight was that there are only nine possible input combinations, ma
     * What is the expected output for the given input?
     * You can use the examples provided in the prompt, or other examples.
 2. Write unit tests for `winner` for the nominal and edge cases you identified in the first step.
+    
+When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs.
```
Comment (yangashley) [inline about-pses/example-pse.md:158]: 👍 How would you feel about adding some examples to this section? Maybe like:

```suggestion
When naming a test, we want to ensure the name describes the scenario we are testing by including information like the function being tested, inputs, and expected outputs. 

For example, if we wanted to test that the function `winner` returns a tie when `player_1` and `player_2` have the same value, then we might name the test something like `test_winner_returns_tie_for_players_with_same_value`.
```

I feel ok leaving this off too because we also have lessons about testing.

File: about-pses/example-pse.md [hunk 4]
```diff
@@ -355,17 +283,17 @@ During an interview, we'd still want to be careful not to spend _too_ much time
 <br>
 
 Example steps:
-1. Check for a tie: 
-    - if player_1 is the same as player_2, return "It's a tie!"
-2. First deal with the case that player_1 is "rock"
+1. First, we'll handle our simplest case which is checking for a tie: 
+    - if `player_1` is the same as `player_2`, return "It's a tie!"
```
Comment (yangashley) [inline about-pses/example-pse.md:288]: I like how it visually looks to have the variables surrounded in tick marks so they appear differently in this long block of text. 

Could we also add tick marks around `player_1` and `player_2` on lines 288-296?

File: about-pses/example-pse.md
Comment (yangashley) [inline about-pses/example-pse.md:355]: I know that this section hasn't been changed in this PR, but how do you feel about providing the link to the gist and then sharing the note about submitting a shareable link?

For me, the flow felt choppy.


```suggestion
Chat Link: [https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1](https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1)

*Note:*
*We are using a gist link in this example to ensure that the link exists from cohort to cohort. When submitting your response, the link should be a shareable link to your chat in the AI tool where you held the conversation.*
```

File: about-pses/example-pse.md (hunk 2 above)
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md]:
```suggestion
1. From the problem description, I observe that...
2. In the example inputs I can see...
3. A question I have around inputs is...
4. Based on the requirements, I am making the assumption...
5. What behavior is expected when...
```

**Review by kelsey-steven-ada — COMMENTED** [review]
Adding other starter suggestions to the questions & observations placeholder

File: about-pses/example-pse.md (hunk 3 above)
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:158]: I feel like including an example here would be a nice reminder and helps guide folks. Would you want to include the example in the question prompt on later PSEs, or just here on the example PSE?

File: about-pses/example-pse.md [hunk 5]
```diff
@@ -355,17 +283,17 @@ During an interview, we'd still want to be careful not to spend _too_ much time
...
     - otherwise player_2 must be "paper", return "Player 2 wins!"
-3. Next deal with the case that player_1 is "paper"
+3. Next we'll deal with the case that player_1 is "paper":
     - if player_2 is "rock", return "Player 1 wins!"
     - otherwise player_2 must be "scissors", return "Player 2 wins!"
-4. Otherwise player_1 must be "scissors"
+4. Otherwise player_1 must be "scissors":
     - if player_2 is "rock", return "Player 2 wins!"
-    - otherwise player_2 must be "paper", return "Player 1 wins!"
+    - otherwise player_2 must be "paper", return "Player 1 wins!"
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md]:
```suggestion
2. If it is not a tie, we'll move through the cases where `player_1` is "rock", "paper", or "scissors" to determine a winner, starting with the case that `player_1` is "rock":
- if `player_2` is "scissors", return "Player 1 wins!"
- otherwise `player_2` must be "paper", return "Player 2 wins!"
3. Next we'll deal with the case that `player_1` is "paper":
- if `player_2` is "rock", return "Player 1 wins!"
- otherwise `player_2` must be "scissors", return "Player 2 wins!"
4. Otherwise `player_1` must be "scissors":
- if `player_2` is "rock", return "Player 2 wins!"
    - otherwise `player_2` must be "paper", return "Player 1 wins!" 
```

File: about-pses/example-pse.md [hunk 6]
```diff
@@ -420,15 +361,19 @@ Chat Link: [https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a
 
 <br>
 
-I received feedback that my approach was clear, and my logical steps that didn’t rely on Python specific syntax, which made it easy to understand. I made good use of logic by handling ties first and explaining each outcome clearly, showing I understood the reasoning behind the rules. 
+One strength the AI highlighted was my use of "otherwise... must be" phrasing throughout my steps. The feedback noted that this showed I understood "that earlier conditions have already eliminated other possibilities," which is the kind of reasoning interviewers want to see. I think this worked well because it demonstrates I wasn't just listing cases mechanically, I was actively narrowing down what remained, which makes the logic easier to follow and verify.
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:370]:
```suggestion
Depending on exactly what the AI shares, a reflection and summary might look like:

<br>

Chat Link: [https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1](https://gist.github.com/ada-instructors/56c872acec770c68a9ae8b81b58a80a1)

<br>

*Note:*

*We are using a gist link in this example to ensure that the link exists from cohort to cohort. When submitting your response, the link should be a shareable link to your chat in the AI tool where you held the conversation.*

<br>

One strength the AI highlighted was my use of "otherwise... must be" phrasing throughout my steps. The feedback noted that this showed I understood "that earlier conditions have already eliminated other possibilities," which is the kind of reasoning interviewers want to see. I think this worked well because it demonstrates I wasn't just listing cases mechanically, I was actively narrowing down what remained, which makes the logic easier to follow and verify.
```

**Review by kelsey-steven-ada — COMMENTED** [review]
Respond to feedback & add updates through suggestions

File: about-pses/example-pse.md (hunk 5 above)
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md]: For some reason GitHub won't allow me to apply this suggestion, so I'll push the change up manually shortly.

File: about-pses/example-pse.md (hunk 3 above)
Comment (yangashley) [reply about-pses/example-pse.md:158]: If we include the example in subsequent question prompts, would they each have to be changed to relevant to the question being asked?

If so, then I think just adding the example in this file is ok.

**Review by yangashley — APPROVED** [review]

File: about-pses/example-pse.md (hunk 3 above)
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:158]: I don't think we need to update the example for each PSE, but we'd likely want to adjust the language to something like:

* e.g. from the example PSE: if we wanted to test that the function `winner` returns a tie when `player_1` and `player_2` have the same value, then we might name the test something like `test_winner_both_inputs_paper_returns_tie`.
